wordpress-mobile/WordPress-Android
# Add reader tests

### Description
This PR adds UI Tests for the Reader screen. 

* `navigateThroughPosts()`: checks a post content and the ability to slide to previous and next posts.
* `likePost()`: checks that a post can be liked and unliked fro post view screen.

**Note 1**: A big part of the tests development was done blindly, in trial and error sessions because Android Emulator in the CI, for unknown reasons, behave different from the Emulator running locally even they being theoretically the same `Pixel 2 API 28`. That's the reason behind the 28 commits.

**Note 2**: This PR has over 3.5K lines added but that's mostly new mocks added.

### Testing Instructions
* The [Connected Tests-2 CI job is green](https://app.circleci.com/pipelines/github/wordpress-mobile/WordPress-Android/28491/workflows/339b0b4f-6604-49b7-b5bd-dc9b36fd4fe5/jobs/118246), and you can see `navigateThroughPosts()`and likePost()` tests included.
* Tests run locally on Pixel 2 API 28 (this is the model used on CI).

## Regression Notes
1. Potential unintended areas of impact
Changes would only affect ui tests.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Checked that all tests are passing locally an on CI.

3. What automated tests I added (or what prevented me from doing so)
* `navigateThroughPosts()`
* `likePost()`

PR submission checklist:
